src/service/action-impl.js

```diff
@@ -71,16 +75,19 @@ export class ActionInvocation {
    * @param {!Node} target
    * @param {string} method
    * @param {?JSONType} args
+   * @param {?string} argsExpression
    * @param {?Element} source
    * @param {?Event} event
    */
-  constructor(target, method, args, source, event) {
+  constructor(target, method, args, argsExpression, source, event) {
```

Comment:
Changed to use `__AMP_OBJECT_LITERAL__`, similar to other constants in this file. IMO this is better than `object-string` since it's clearer to users that this isn't a public API.

> If we do use that, we should absolutely add a test that proves the parser will throw trying to use it outside of this object literal codepath.

Not sure we can test this since it's possible for users to create actions like `on="target.event(__AMP_OBJECT_LITERAL__=123)`. Code to enforce this specifically at the parser-level would be even uglier than the previous `argsExpression` approach due to loss of generality.